test/SemanticVersioning.Tests/PreReleaseRanges.cs

```diff
@@ -67,6 +66,7 @@ public void MatchingPreReleaseWithIncludePrereleases(string rangeString, string
         [InlineData("<1.3", "1.3.0-alpha.1")]
         [InlineData("~1.2.3", "1.3.0-alpha.1")]
         [InlineData("^0.2.3", "0.2.2-alpha")]
+        [InlineData("^0.2.3", "0.2.3-alpha")]
```

Comment:
I think this change is correct, and matches the reasoning in https://github.com/npm/node-semver/issues/409, but node-semver (version 7.5.0) actually treats this as satisfied. It looks like this is possibly a bug in node-semver though as this behaviour is inconsistent compared to versions with a major version number:
```
> semver.satisfies('0.2.3-alpha', '^0.2.3', {includePrerelease: true})
true
> semver.satisfies('1.2.3-alpha', '^1.2.3', {includePrerelease: true})
false
```

Reply:
I've raised https://github.com/npm/node-semver/issues/557 for this, will hold off merging this for now and see if I get a response.